src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationRoot.cs

```diff
@@ -104,9 +104,10 @@ public void Dispose()
                 registration.Dispose();
             }
 
-            // dispose providers
+            // dispose providers and the sources
             foreach (IConfigurationProvider provider in _providers)
             {
+                (provider.ConfigurationSource as IDisposable)?.Dispose();
```

Comment:
the root has no reference sources, but it can now access `provider.ConfigurationSource`  and dispose when possible